cumulusci/core/config/base_config.py

```diff
@@ -23,7 +30,47 @@ def _load_config(self):
         """Subclasses may override this method to initialize :py:attr:`~config`"""
         pass
 
+    @classmethod
+    def allowed_names(cls) -> set:
+        properties = set(dir(cls))
+        annotations = set(getattr(cls, "__annotations__", {}))
+        return properties.union(annotations)
+
+    # long term plan is to get rid of this
     def __getattr__(self, name):
+        if not name.startswith("_"):
+            first_part = name.split("__")[0]
+            if (
+                first_part
+                not in self.all_allowed_names()
+                # and first_part not in self.config
+            ):
+                warnings.warn(
+                    f"__getattr__ on Configs is deprecated: `{first_part}` on `{self.__class__.__name__}`",
+                    DeprecationWarning,
+                )
+
+                assert (
+                    not STRICT_GETATTR
+                ), f"__getattr__ on Configs is deprecated: `{first_part}` on `{self.__class__.__name__}`"
+        return self.lookup(name)
+
+    @classmethod
+    @lru_cache
+    def all_allowed_names(cls):
```

Comment:
Probably best to call them _allowed_names and _all_allowed_names

Reply:
Fixed in ee235fe01